### PR TITLE
Workflow tutorial

### PR DESCRIPTION
**Pull request by priyanshunayan**
I tried to make this article as accessible as possible for everyone. But I found myself using the terms `json-ld`, `web APIs`, `hydra` again and again. I did not want to overwhelm beginners with the comprehensive guides about these terms, so I included bare minimum information about this technology, just enough to follow along with the tutorial. Please let me know if this needs addtions/subtraction.

[EDIT] - Page to review: https://deploy-preview-3--hydra-docs.netlify.app/src-content-tutorial-workflow

### CODE REVIEW

**Comment by king-11** [comment]
can we use a .webp image instead of .png 🤔

**Comment by Purvanshsingh** [comment]
We can add hyperlinks to corresponding Repos,
hydra-python-core repo can be hyperlinked in **Defining the API Documentation** Section,
hydrus repo in **Passing the API documentation to hydrus** Section, and
hydra-python-agent repo in **Using smart Clients to consume APIs** Section.

File: docs/src/content/tutorial/workflow.md [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+<div id="main">
+After going through this document, you should be able to understand the general workflow followed to create and use APIs using the tools of the Hydra Ecosystem. Though the workflow might be different for few use cases, this document tries to cover the general workflow that can satisfy most of the use cases.
+</div>
+
+To design web APIs using the tools of this ecosystem, follow three simple steps.
+
+1. Define the API in JSON-LD format as API Documentation
+2. Pass this API Documentation to the `hydrus` server
+3. Use Smart clients(`hydra-python-agent`) to consume APIs served by the `hydrus`.
+
```
Comment (sameshl) [inline docs/src/content/tutorial/workflow.md:72]: Hmm, when we are keeping the heading of this article is `General Workflow`, I think the reader should directly get access to the workflow without first getting interrupted by having to go through the related terminologies.

I understand that these terminologies are very important to understand the workflow so how about we probably move all these to another page(which has only terminologies/glossary) which we can hyperlink here?

File: docs/src/content/tutorial/workflow.md [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+It changes the existing APIDoc in JSON-LD to Python Classes which the tools like `hydra-python-agent` and `hydrus` can understand.
+
+## 2. Passing the API documentation to `hydrus`
+
+`hydrus` accepts the APIDoc as a parameter and spins up a server, and exposes routes for every Hydra Class and Collection with the allowed operations as defined in the APIDoc. One can configure the authentication mechanism, and the database.
+
+Hydrus can be used a Command Line Interface(CLI) and as a python package.
+
+## 3. Using smart Clients to consume APIs
```
Comment (sameshl) [inline docs/src/content/tutorial/workflow.md:91]: Just a minor nitpick, I think we should change `Clients` to `clients`.

**Comment by sameshl** [comment]
Overall, a very good article @priyanshunayan! :+1:

**Comment by chrizandr** [comment]
Made a new workflow diagram. 
Workflow

Editable link, also availble if you need to modify anything. @priyanshunayan 
https://drive.google.com/file/d/1yvqx9OxIPeVKPZwaIFIQV0WolLyMj9vc/view?usp=sharing

File: docs/src/content/tutorial/workflow.md (hunk 1 above)
Comment (priyanshunayan) [reply docs/src/content/tutorial/workflow.md:72]: The very first paragraph contains the note with a link to the main content, if they are familiar with these, they can jump directly there. That being said, I was not convinced of the idea to direct people to go through a glossary first. I think that will be a little overwhelming. Better to wait to see what other reviewers think about this!

**Comment by priyanshunayan** [comment]
@chrizandr This is a wonderful illustration. Much Much better than what I had put. This made me understand the whole workflow document at a glance! 

Just like hydra-python-agent, hydrus can also be used as a python package, as well as a CLI, did you hide that purposefully? I can add that if you missed it?

**Comment by chrizandr** [comment]
hydrus CLI is just to start the server right? I don't think that would be needed. We're not using that to access data. It would be great if we could add something to show crud operations. Right now it just looks like client is consuming data from hydrus but nothing shows that client can also add data.

**Comment by Mec-iS** [comment]
in perspective everything useful for the developer should be available via CLU, examples: bulk-load data from common format, delete instances and relationships, etc. obviously these operations should use the REST API to apply the changre

**Comment by Purvanshsingh** [comment]
I have an idea to create a separate repo with Movie API implementation in which we can show both adding and retrieving data.
It will be a demo repo similar to hydra-flock-drone, But I think this will make a more clear picture of how Hydra works.

> hydrus CLI is just to start the server right? I don't think that would be needed. We're not using that to access data. It would be great if we could add something to show crud operations. Right now it just looks like client is consuming data from hydrus but nothing shows that client can also add data.

**Review by Mec-iS — COMMENTED** [review]
It is important to have uniform treatment for names, as engineers use standard names to point to precise concepts and objects:
* json is not JSON (as it is an acronym)
* Linked Data (W3C) is a different object than just "linked data" (general)
* writing "hydra" is an error as the official standard specification name is Hydra
* etc

I know this is coming from old contributions and it is time to address this non-conformity.
If you have any doubt about the context and which name to use, please ask.

**Comment by Purvanshsingh** [comment]
There is a small typo 

Screenshot from 2021-01-20 10-39-52

**Comment by chrizandr** [comment]
- Other two methods for creating hydra doc need to be added
- Little more explaininig about linked data, what are classes, props, operations, how do these concepts connect to API
- API doc definition
- How does hydrus connect to client, brief
- GUI is used to interact with server through agent.

**Pull request merged**